lib/auth/signInWithGithub.ts

```diff
@@ -10,7 +10,7 @@ export default async function signInWithGithub() {
   const { data, error } = await supabase.auth.signInWithOAuth({
     provider: "github",
     options: {
-      redirectTo: getCallbackUrl(),
+      redirectTo: `${getCallbackUrl()}?next=apply`,
```

Comment:
This is usually safe; I think we are only working with data we trust, but we should really avoid stuff like this. Unthinkingly concatenating can lead to all sorts of problems and is thus an antipattern. 

I would take a look at the [URL API](https://developer.mozilla.org/en-US/docs/Web/API/URL_API) and use its [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams)  interface to more safely to this.

### example in my node repl

```
Welcome to Node.js v22.3.0.
Type ".help" for more information.
> url_str = "https://example.com"
'https://example.com'
> url = new URL(url_str)
URL {
  href: 'https://example.com/',
  origin: 'https://example.com',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'example.com',
  hostname: 'example.com',
  port: '',
  pathname: '/',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
> url.searchParams.set("next", "apply")
undefined
> url.toString()
'https://example.com/?next=apply'
```